### PR TITLE
fix: estimate gas price from signer LF-2417

### PR DESCRIPTION
**Pull request by Abhikumar98**
# FIX

Handle Transaction underpriced error [LF-2417]

## TL;DR
Automatically handle "Transaction Underpriced" errors by retrying the transaction with an increased gas price.

## DESCRIPTION
This feature handles the scenario where a transaction fails with a "Transaction Underpriced" error. Upon encountering such an error, the feature will automatically retry the transaction with an increased gas price. The latest gas price will be fetched from the signer, and it will be increased by 25% before sending the transaction forward.

## STEPS TO TEST
1. Initiate a swap or bridge transaction.
2. While at the Metamask screen, reduce the Max Base Fee and Priority Fee
3. Confirm the transaction and observe the "Transaction Underpriced" error.
4. Upon the next retry the transaction will be retried with an increased gas price.
5. Verify that the retried transaction is sent with a gas price estimated from the signer and increased by 25%.

## EXPECTED RESULTS
Upon encountering a "Transaction Underpriced" error, the feature should automatically retry the transaction with an increased gas price based on the latest signer estimate.

## CHECKLIST
- [x] I have performed a self-review of my code.
- [x] I have commented the code I wrote to make it easily understandable.
- [x] I have added tests that cover the new functionality.
- [x] I have ensured that existing tests are passing.

[LF-2417]: https://lifi.atlassian.net/browse/LF-2417?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Abhikumar98** [comment]
Avoiding the default increase in gas price in StepExecutionManager.ts as it'd always result in an increase gas price.

File: src/execution/StepExecutionManager.ts [hunk 1]
```diff
@@ -163,6 +163,13 @@ export class StepExecutionManager {
                 (BigInt(estimatedGasLimit.toString()) * 125n) / 100n
               }`
             }
+
+            // Fetch latest gasPrice from provider and use it
+            const gasPrice = await signer.getGasPrice()
```
Comment (chybisov) [inline src/execution/StepExecutionManager.ts:168]: If this fails the whole swap via the widget will throw an unknown error. We should wrap all calls to `signer.getGasPrice` and `signer.estimateGas` try...catch and fallback to BE values/previous logic.

**Review by chybisov — CHANGES_REQUESTED** [review]

File: src/execution/StepExecutionManager.ts (hunk 1 above)
Comment (chybisov) [reply src/execution/StepExecutionManager.ts:168]: I got an error while testing 😄

Comment (Abhikumar98) [reply src/execution/StepExecutionManager.ts]: Got it. I've pushed the changes

**Review by Addminus — APPROVED** [review]
lets go
